streams/src/main/java/org/apache/kafka/streams/processor/StandbyUpdateListener.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor;
+
+import org.apache.kafka.common.TopicPartition;
+
+public interface StandbyUpdateListener {
+
+    enum SuspendReason {
+        MIGRATED,
+        PROMOTED
+    }
+
+    /**
+     * Method called upon the creation of the Standby Task.
+     *
+     * @param topicPartition   the TopicPartition of the Standby Task.
+     * @param storeName        the name of the store being watched by this Standby Task.
+     * @param startingOffset   the offset from which the Standby Task starts watching.
+     * @param currentEndOffset the current latest offset on the associated changelog partition.
```

Comment:
```suggestion
     * @param currentEndOffset the current end offset on the associated changelog partition.
```
I guess you could also say "highest offset", but "latest" feels a bit ambiguous, at least to me personally